### PR TITLE
Fix area defs in cmsaf-claas2 reader

### PR DESCRIPTION
**Pull request by sfinkens**
Area extents in CM SAF CLAAS-2 Level 2 data are affected by the following two problems

- They do not take the georeferencing offset (before Dec 2017) into account
- They are incorrect after 2016

Fix those problems by using pre-defined area defs (with and without offset) and ignore the extents from the file.

TODO

- [x] Refactor tests: Write and read small test file
- [x] Add area defs

### CODE REVIEW

File: satpy/readers/cmsaf_claas2.py [hunk 1]
```diff
@@ -74,11 +95,14 @@ def get_dataset(self, dataset_id, info):
...
         """Get the area definition."""
-        return pyresample.geometry.AreaDefinition(
-                "some_area_name",
-                "on-the-fly area",
-                "geos",
-                self["/attr/CMSAF_proj4_params"],
-                self["/dimension/x"],
-                self["/dimension/y"],
-                self["/attr/CMSAF_area_extent"])
+        full_disk = self._get_full_disk()
```
Comment (mraspaud) [inline satpy/readers/cmsaf_claas2.py:98]: I would put this in `_subset_full_disk`, what do you think?

**Review by mraspaud — APPROVED** [review]
LGTM, very clean tests, thanks for fixing them too!
Just one very tiny comment inline

File: satpy/readers/cmsaf_claas2.py (hunk 1 above)
Comment (sfinkens) [reply satpy/readers/cmsaf_claas2.py:106]: Good idea, thanks!

**Review by mraspaud — APPROVED** [review]
LGTM

**Pull request merged**